### PR TITLE
feat: add SEO reviewer with automated content updates

### PR DESCRIPTION
**Pull request by katieschilling**
## Summary

Adds a comprehensive SEO reviewer agent that analyzes blog posts and suggests improvements for SEO optimization, LLM optimization, and brand consistency. The tool can automatically apply safe changes like meta descriptions, tag management, and alt text from captions.

## Key Features

- **SEO Analysis**: Title length, meta descriptions, keywords, content structure, image alt text
- **LLM Optimization**: Question-style headings, problem/solution structure, technical context
- **Tigris Terminology**: Brand consistency, preferred descriptions, technical accuracy
- **Smart Tag Suggestions**: Content-based tag recommendations with consistency checks
- **Automated Updates**: Safe application of meta descriptions, tag fixes, caption-to-alt-text
- **Security**: File path validation, content size limits, atomic operations

## Usage

```bash
# Review posts
npm run seo:review
npm run seo:review:all

# Apply automated fixes
npm run seo:apply
npm run seo:apply:all

# Direct usage
node seo-reviewer.js [file] [--apply] [--all]
```

## What Gets Automated vs Manual Review

**✅ Automated with --apply:**
- Meta description improvements
- Tag additions and casing fixes (ai → AI)
- Category tag reordering
- Caption text as alt text
- Deprecated term replacement (cross-cloud → multi-cloud)

**❌ Manual review required:**
- Missing titles and content structure
- Image alt text (when no captions)
- Content length and heading improvements
- Complex terminology decisions

## Test Plan

- [x] Test SEO analysis on existing blog posts
- [x] Verify automated changes are applied safely
- [x] Test security features (path validation, file limits)
- [x] Validate npm scripts work correctly
- [x] Test both individual and bulk analysis
- [x] Verify caption detection and alt text application
- [x] Test error handling and edge cases

🤖 Generated with Claude Code

### CODE REVIEW

File: seo-reviewer.mjs [hunk 1]
```diff
@@ -0,0 +1,2015 @@
...
+    // Handle simple arrays [item1, item2]
+    if (value.startsWith("[") && value.endsWith("]")) {
+      return value
+        .slice(1, -1)
+        .split(",")
+        .map((item) => item.trim());
+    }
+
+    return value;
+  }
```
[the remark this answers is not in the document]
Comment (katieschilling) [reply seo-reviewer.mjs:157]: use greymatter instead of regex to parse the frontmatter of the post

**Review by Xe — APPROVED** [review]
